go/protomodule/protomodule_message_test.go

```diff
@@ -966,3 +967,82 @@ func TestProtoMergeDiffTypes(t *testing.T) {
 		t.Errorf("expected error %q, got %q", errorMsg, err.Error())
 	}
 }
+
+// Pre 1.0 Skycfg allowed maps to be constructed with None values for proto2 (see protoMap.SetKey)
+func TestMapNoneCompatibility(t *testing.T) {
+	val, err := evalFunc(`
+def fun():
+    pb = proto.package("skycfg.test_proto")
+    msg = pb.MessageV2()
+    m = {
+	    "a": pb.MessageV2(),
+	    "b": pb.MessageV2(),
+	    "c": pb.MessageV2(),
+	    "d": None,
+    }
+    msg.map_submsg = m
+
+    m2 = msg.map_submsg
+    m2["b"] = None
+    m2.setdefault("e", None)
+    m2.update([("c", None)])
+
+    return msg
+`, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	got := mustProtoMessage(t, val).(*pb.MessageV2)
+
+	checkProtoEqual(t, &pb.MessageV2{
+		MapSubmsg: map[string]*pb.MessageV2{
+			"a": &pb.MessageV2{},
+		},
+	}, got)
+
+	// Confirm this only works for all in proto2, only message values in proto3
```

Comment:
tbh, I'm not sure if I'm able to parse this comment (line 1003) correctly, but I'm assuming it says something like "confirm that setting a map value to None works for proto2 but not for proto3"?

Reply:
I believe the intent in the previous implementation is that scalar and message values can be set to none in proto2, but in proto3 only setting messages to none is allowed (equivalent to unsetting).

With the behavior now being to unset the fields, it seems sort of arbitrary to allow it in proto2 versus proto3 but just trying to capture the original behavior. IMO ideally we wouldn't support setting none in maps to unset